src/vcpkg/archives.cpp

```diff
@@ -177,6 +187,31 @@ namespace
 
 namespace vcpkg
 {
+    void extract_self_extracting_7z(const VcpkgPaths& paths, const Path& archive, const Path& to_path)
+    {
+        // static_cast due to Warning C4309
+        constexpr static const char header_7z[] = {
+            '7', 'z', static_cast<char>(0xbc), static_cast<char>(0xaf), 0x27, 0x1c};
+
+        const Path stem(archive.stem());
```

Comment:
```suggestion
        const Path stem = archive.stem();
```

whenever possible we should avoid direct initialization imo.

Reply:
why?